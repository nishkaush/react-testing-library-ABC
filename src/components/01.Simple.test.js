//4 things to import

import React from "react";
import { render, queryAllByText, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Simple from "./01.Simple";

// snapshot testing
// assertions on component

it("renders", () => {
  const { asFragment } = render(<Simple text="yahoo" />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const { getByText } = render(<Simple text="yahoo" />);
  const elm = getByText("yahoo");
  // const elm = queryAllByText("yahoo");
  expect(elm).toBeInTheDocument();
});

it("shows input element", () => {
  const { getByDisplayValue } = render(<Simple text="yahoo" />);
  expect(getByDisplayValue("yaddi")).toBeInTheDocument();
});

// GetBy + Text

// QueryBy + PlaceholderText --> negative situation
