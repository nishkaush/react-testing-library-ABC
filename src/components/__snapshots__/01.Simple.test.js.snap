// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <h1>
      yahoo
    </h1>
    <input
      type="text"
      value="yaddi"
    />
  </div>
</DocumentFragment>
`;
